src/Jank/TermManagement/TermList.tsx

```diff
@@ -1,18 +1,29 @@
-import React from 'react';
-import Term from './Term';
-import useApi from '../../hooks/useApi';
-import { AddButton, AddLabel, TermInput } from './ui';
+import React, { useEffect, useReducer } from 'react';
 
-type Terms = {
-  _id: string,
-  value: string
-}
+import useApi from '../../hooks/useApi';
+import reducer from './reducer';
+import TermItem from './Term';
+import { Term } from './types';
+import {
+  AddButton,
+  AddLabel,
+  ErrorText,
+  TermInput,
+} from './ui';
+import { setTerms } from './actions';
 
 const TermList = () => {
-  const [error, terms] = useApi<Terms[]>('/jank/terms');
+  const [error, apiTerms] = useApi<Term[]>('/jank/terms');
+  const [terms, dispatch] = useReducer(reducer, []);
+
+  useEffect(() => {
+    if (apiTerms !== null) {
+      dispatch(setTerms(apiTerms));
```

Comment:
As we cannot initialize the reducer with the list of terms right away (we need to fetch them from the server first), we dispatch an action here to set the terms asynchronously, once we receive a response from the server. This will automatically cause a re-render when that happens.